### DISCUSSION TITLE
need help to understand error "ClosedSessionException: acquired an unhealthy connection"

### DISCUSSION THREAD

**Discussion by billshen99** [line/armeria#5270] [Q&A]
Hey folks, I'm running a gprc service through armeria in Kotlin. Recently, we start to see a lot of errors when calling server sometimes. 
error message: `com.linecorp.armeria.client.UnprocessedRequestException: com.linecorp.armeria.common.ClosedSessionException: acquired an unhealthy connection`

We saw the retry happens, but it didn't resolve the issue and connect to server only for partial requests. Some requests still can reach the server successfully. we mitigate the issue by bouncing the pods manually.  can anyone help me to understand in what possible situations it can trigger `acquired an unhealthy connection` and why retry cannot resolve this issue? 

Thanks.

**Comment by minwoox** [comment]
Hi, that's very interesting.
I've never seen this before so I need more information.
- Which Armeria version are you using?
- Which HTTP version are you using?
- Does this happen suddenly or did you upgrade Armeria recently?
- Could you also provide us a reproducible example if possible?
